### ISSUE TITLE
BTVenmoClient `tokenize` async method leaks memory and does not return/throw an error when the user cancels the authorisation.

### ISSUE THREAD

**Issue by mdeaconu** [braintree/braintree_ios#1085]
### Braintree SDK Version

6.4.0

### Environment

Both

### Xcode Version

Xcode 14.3.1

### OS Version & Device

iOS 16.2

### Integration type

Swfit Package Manager

### Development Processor

Intel

### Describe the bug

Given the sample provided below when the user cancels the authorisation within the Venmo app the `tokenize` method call does not return an error or at least something that indicates the cancelation.

```
venmoClient = BTVenmoClient(apiClient: btApiClient)
let request = BTVenmoRequest(paymentMethodUsage: .multiUse)
do {
    let nonceAccount = try await venmoClient.tokenize(request)
    // send nonceAccount.nonce
} catch let error {
    // handle error here
}
```

### To reproduce

1. Create an `BTVenmoRequest`
2. Call the `tokenize` async method
3. Cancel the authorisation from Venmo app.
4. Check the return nonce / or error

### Expected behavior

I would expect to get an error code that would indicate the cancelation.

### Screenshots

<img width="279" alt="Screenshot 2023-08-01 at 17 09 58" src="https://github.com/braintree/braintree_ios/assets/2821200/93bacc4d-99cc-4f30-8c25-1829fd3768cc">
<img width="558" alt="Screenshot 2023-08-01 at 17 15 30" src="https://github.com/braintree/braintree_ios/assets/2821200/bd296263-e807-430b-af96-d1983eb89652">

**Comment by scannillo** [comment]
👋 Hi @mdeaconu - thank you for bringing this issues to us. I am trying to replicate it now.

Can you provide us with the URL that you're getting back in your scene or app delegate that you then pass to one of [BTAppContextSwitcher's handleOpenURL methods](https://github.com/braintree/braintree_ios/blob/b24166809d095d28ecaf471292f0d2a79ee1e652/Sources/BraintreeCore/BTAppContextSwitcher.swift#L26-L43)?

Also if you have screenshots or videos of what the specific cancelation flow you're triggering from the Venmo app, that would be helpful.

**Comment by scannillo** [comment]
We are in the process of fixing the cancel case that you mentioned and will link that PR soon.

In the meantime, we were not able to replicate a memory leak like you showed. You should check on your `VenmoTokenizationHandler`.

Feel free to test on [this branch we made](https://github.com/braintree/braintree_ios/tree/venmo-demo-swift) which uses the async Swift function for `BTVenmoClient.tokenize()` in our Demo app ([see specific line here](https://github.com/braintree/braintree_ios/blob/4041d3eb9fa92d41fd201391b5b990cf40adfd5b/Demo/Application/Features/Venmo%20-%20Custom%20Button/BraintreeDemoVenmoViewController.swift#L79)). Using this Demo app, we aren't able to replicate a leak of `BTVenmoClient`.

**Comment by mdeaconu** [comment]
> We are in the process of fixing the cancel case that you mentioned and will link that PR soon.
> 
> In the meantime, we were not able to replicate a memory leak like you showed. You should check on your `VenmoTokenizationHandler`.
> 
> Feel free to test on [this branch we made](https://github.com/braintree/braintree_ios/tree/venmo-demo-swift) which uses the async Swift function for `BTVenmoClient.tokenize()` in our Demo app ([see specific line here](https://github.com/braintree/braintree_ios/blob/4041d3eb9fa92d41fd201391b5b990cf40adfd5b/Demo/Application/Features/Venmo%20-%20Custom%20Button/BraintreeDemoVenmoViewController.swift#L79)). Using this Demo app, we aren't able to replicate a leak of `BTVenmoClient`.

Hello,
`VenmoTokenizationHandler` class acts similar to `BraintreeDemoVenmoViewController` which keeps a strong ref to `BTVenmoClient`. Normally it should go away and release memory once `tokenize()` call finishes.

Using the branch specified I was able to create some leaks:
```
2023-08-02 18:26:58.227297+0300 Demo[34922:145802] Fetching Client Token…
2023-08-02 18:27:00.933036+0300 Demo[34922:145802] Loading integration: BraintreeDemoVenmoViewController
2023-08-02 18:27:01.190886+0300 Demo[34922:145802] Using Client Token
2023-08-02 18:27:01.191329+0300 Demo[34922:145802] Presenting Demo.BraintreeDemoVenmoViewController
2023-08-02 18:27:03.457479+0300 Demo[34922:145802] Tapped Venmo - initiating Venmo auth
2023-08-02 18:27:13.406876+0300 Demo[34922:145802] Tapped Venmo ECD - initiating Venmo auth
SWIFT TASK CONTINUATION MISUSE: tokenize(_:) leaked its continuation!
2023-08-02 18:27:14.050144+0300 Demo[34922:145802] SWIFT TASK CONTINUATION MISUSE: tokenize(_:) leaked its continuation!
2023-08-02 18:40:22.739811+0300 Demo[34922:145802] Fetching Client Token…
2023-08-02 18:40:29.207586+0300 Demo[34922:145802] Loading integration: BraintreeDemoVenmoViewController
2023-08-02 18:40:29.466513+0300 Demo[34922:145802] Using Client Token
2023-08-02 18:40:29.466871+0300 Demo[34922:145802] Presenting Demo.BraintreeDemoVenmoViewController
2023-08-02 18:40:31.870273+0300 Demo[34922:145802] Tapped Venmo - initiating Venmo auth
2023-08-02 18:40:44.188042+0300 Demo[34922:145802] Tapped Venmo - initiating Venmo auth
SWIFT TASK CONTINUATION MISUSE: tokenize(_:) leaked its continuation!
2023-08-02 18:40:45.761560+0300 Demo[34922:145802] SWIFT TASK CONTINUATION MISUSE: tokenize(_:) leaked its continuation!
2023-08-02 18:40:52.471816+0300 Demo[34922:145802] Tapped Venmo - initiating Venmo auth
SWIFT TASK CONTINUATION MISUSE: tokenize(_:) leaked its continuation!
2023-08-02 18:40:53.112100+0300 Demo[34922:145802] SWIFT TASK CONTINUATION MISUSE: tokenize(_:) leaked its continuation!
```

Steps to reproduce:
1. Tap on `Settings`
2. Select `Venmo - Custom Button` Integration
3. Tap on `Done`
4. Tap several times on `Venmo` button (used MockVenmo sample app available)
5. Repeat step 1-4

However, I used this branch `venmo-cancel-error` and it looks the issue got fixed.

Will the change/fix be part of the next release? I am trying just to get an ETA.

Thanks!

**Comment by jaxdesmarais** [comment]
Hello @mdeaconu -

Glad to hear the new branch/PR is resolving the memory leak issue you saw previously. This change will go into our next release of the SDK and we will update this thread with the released version once completed.

**Comment by jaxdesmarais** [comment]
Hello @mdeaconu -

This was released in [version 6.5.0](https://github.com/braintree/braintree_ios/releases/tag/6.5.0). Please let us know if you run into any issues after making the update. Thanks again for bringing this issue to our attention! 🚀
